src/NServiceBus.RavenDB/Outbox/RavenDbOutboxStorage.cs

```diff
@@ -0,0 +1,110 @@
+﻿namespace NServiceBus.RavenDB.Outbox
+{
+    using System;
+    using System.Configuration;
+    using System.Threading;
+    using NServiceBus.Features;
+    using NServiceBus.Outbox.RavenDB;
+    using NServiceBus.RavenDB.Internal;
+    using Raven.Client;
+
+    class RavenDbOutboxStorage : Feature
+    {
+        public RavenDbOutboxStorage()
+        {
+            DependsOn<Outbox>();
+            DependsOn<SharedDocumentStore>();
+            RegisterStartupTask<OutboxCleaner>();
+        }
+
+        protected override void Setup(FeatureConfigurationContext context)
+        {
+            var store =
+                // Trying pulling a shared DocumentStore set by the user or other Feature
+                context.Settings.GetOrDefault<IDocumentStore>(RavenDbSettingsExtensions.DocumentStoreSettingsKey) ?? SharedDocumentStore.Get(context.Settings);
+
+            if (store == null)
+            {
+                throw new Exception("RavenDB is configured as persistence for Outbox and no DocumentStore instance found");
+            }
+
+            ConnectionVerifier.VerifyConnectionToRavenDBServer(store);
+
+            Helpers.SafelyCreateIndex(store, new OutboxRecordsIndex());
+
+            context.Container.ConfigureComponent<Installer>(DependencyLifecycle.InstancePerCall)
+                .ConfigureProperty(c => c.StoreToInstall, store);
+
+            context.Container.ConfigureComponent<OutboxPersister>(DependencyLifecycle.InstancePerCall)
+                .ConfigureProperty(x => x.DocumentStore, store);
+
+            context.Container.ConfigureComponent<OutboxRecordsCleaner>(DependencyLifecycle.InstancePerCall)
+                .ConfigureProperty(x => x.DocumentStore, store);
+        }
+
+        class OutboxCleaner : FeatureStartupTask, IDisposable
+        {
+            Timer cleanupTimer;
+            TimeSpan frequencyToRunDeduplicationDataCleanup;
+            TimeSpan timeToKeepDeduplicationData;
+            public OutboxRecordsCleaner Cleaner { get; set; }
+
+            public void Dispose()
+            {
+                if (cleanupTimer != null)
+                {
+                    cleanupTimer.Dispose();
+                    cleanupTimer = null;
+                }
+            }
+
+            protected override void OnStart()
+            {
+                var configValue = ConfigurationManager.AppSettings.Get("NServiceBus/Outbox/RavenDB/TimeToKeepDeduplicationData");
```

Comment:
No not really. Don't worry I'll take care of it

-- 

Daniel Marbach

Solution Architect, Particular Software

Email:  daniel.marbach@particular.net

Web:    www.particular.net
<http://t.signauxneuf.com/e1t/c/5/f18dQhb0S7lC8dDMPbW2n0x6l2B9nMJN7t5XX4RzDjlVS9QB88pTc_dW3Ljg7x56dWyBdgBXs402?t=http%3A%2F%2Fwww.particular.net%2F&si=4896787407044608&pi=33cc6202-1f31-4040-8d6c-e70641f44bba>